compiler/z/codegen/OMRMemoryReference.cpp

```diff
@@ -3696,6 +3696,12 @@ generateS390MemoryReference(TR::Register * br, int32_t disp, TR::CodeGenerator *
 TR::MemoryReference *
 generateS390MemoryReference(TR::Node * node, TR::CodeGenerator * cg, bool canUseRX)
    {
+   // The TR::MemoryReference(TR::Node*, TR::CodeGenerator*, bool); constructor (below) is made for store or load nodes
+   // that have symbol references.
+   // A symbol reference is needed to adjust memory reference displacement (TR::MemoryReference::calcDisplacement).
+   // If the node has no sym ref, this memory reference's gets a NULL symbol reference, which can lead to crashes in displacement
+   // calculations.
```

Comment:
Based on the comment, would it make more sense to push the TR_ASSERT to the constructor instead?

Reply:
This has been done.